packager/hls/base/media_playlist.cc

```diff
@@ -599,6 +601,10 @@ void MediaPlaylist::AddSegmentInfoEntry(const std::string& segment_file_name,
           << start_time << ".";
       entries_.emplace_back(new DiscontinuityEntry());
     }
+  } else if (hls_params_.media_sequence_number > 0 &&
+    hls_params_.media_sequence_number == media_sequence_number_) {
+    // When there's a forced media_sequence_number, start with discontinuity.
+    entries_.emplace_back(new DiscontinuityEntry());
```

Comment:
It looks like we can move this block to the constructor, i.e. after line 347, which simplifies the logic a little bit:

      group_id_(group_id),
      media_sequence_number_(hls_params_.media_sequence_number) {
  // When there's a forced media_sequence_number, start with discontinuity.
  if (media_sequence_number_ > 0)
    entries_.emplace_back(new DiscontinuityEntry());
}